cmd/celestia/blob.go

```diff
@@ -0,0 +1,236 @@
+	Short: "Returns the blob for the given namespace by commitment at a particular height.",
+	Run: func(cmd *cobra.Command, args []string) {
+		num, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			fmt.Fprintln(os.Stderr,
+				fmt.Errorf("error parsing height: uint64 could not be parsed"),
+			)
+			os.Exit(1)
+		}
+
+		// 2. Namespace
+		namespace, err := parseV0Namespace(args[1])
+		if err != nil {
+			fmt.Fprintln(os.Stderr,
+				fmt.Errorf("error parsing namespace: %v", err),
+			)
+			os.Exit(1)
+		}
+
+		// 3: Commitment
+		commitment, err := base64.StdEncoding.DecodeString(args[2])
+		if err != nil {
+			fmt.Fprintln(os.Stderr,
+				errors.New("error decoding commitment: base64 string could not be decoded"),
+			)
+			os.Exit(1)
+		}
+
+		blob, err := rpcClient.Blob.Get(cmd.Context(), num, namespace, commitment)
+		if err == nil {
+			if data, decodeErr := tryDecode(blob.Data); decodeErr == nil {
+				blob.Data = data
+			}
+		}
+
+		output := prepareOutput(blob, err)
+		fmt.Fprintln(os.Stdout, string(output))
+	},
+}
+
+var getAllCmd = &cobra.Command{
+	Use:   "GetAll [params]",
+	Args:  cobra.ExactArgs(2),
+	Short: "Returns all blobs for the given namespace at a particular height.",
+	Run: func(cmd *cobra.Command, args []string) {
+		num, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			fmt.Fprintln(os.Stderr,
+				fmt.Errorf("error parsing height: uint64 could not be parsed"),
+			)
+			os.Exit(1)
+		}
+
+		// 2. Namespace
+		namespace, err := parseV0Namespace(args[1])
+		if err != nil {
+			fmt.Fprintln(os.Stderr,
+				fmt.Errorf("error parsing namespace: %v", err),
+			)
+			os.Exit(1)
+		}
+
+		blobs, err := rpcClient.Blob.GetAll(cmd.Context(), num, []share.Namespace{namespace})
+		if err == nil {
+			for _, b := range blobs {
+				if data, err := tryDecode(b.Data); err == nil {
+					b.Data = data
+				}
+			}
+		}
+
+		output := prepareOutput(blobs, err)
+		fmt.Fprintln(os.Stdout, string(output))
+	},
+}
+
+var submitCmd = &cobra.Command{
+	Use:   "Submit [params]",
+	Args:  cobra.ExactArgs(2),
+	Short: "Submit the blob at the given namespace. Note: only one blob is allowed to submit through the RPC.",
+	Run: func(cmd *cobra.Command, args []string) {
+		// 1. Namespace
+		namespace, err := parseV0Namespace(args[0])
+		if err != nil {
+			fmt.Fprintln(os.Stderr,
+				fmt.Errorf("error parsing namespace: %v", err),
+			)
+			os.Exit(1)
+		}
+
+		// 2. Blob data
+		var blobData []byte
+		switch {
+		case strings.HasPrefix(args[1], "0x"):
+			decoded, err := hex.DecodeString(args[1][2:])
+			if err != nil {
+				fmt.Fprintln(os.Stderr, errors.New("error decoding blob: hex string could not be decoded"))
+				os.Exit(1)
+			}
+			blobData = decoded
+		case strings.HasPrefix(args[1], "\""):
+			// user input an utf string that needs to be encoded to base64
+			src := []byte(args[1])
+			blobData = make([]byte, base64.StdEncoding.EncodedLen(len(src)))
+			base64.StdEncoding.Encode(blobData, []byte(args[1]))
```

Comment:
I am not sure we need to encode if it will be encoded by JSON. I think the original purpose was to encoded to make it compatible with the server side which auto-decoded, but now it's done automatically. We can also likely remove tryDecode

Reply:
No, without tryDecode we will still received unexpected format